typegen/src/typegen/validation.rs

```diff
@@ -67,11 +72,8 @@ pub fn validate_substitutes_and_derives_against_registry(
 }
 
 /// Checks if a given type path is the type path of a type in the registry.
-pub fn registry_contains_type_path(types: &PortableRegistry, path: &syn::Path) -> bool {
-    let scale_type_path = scale_info::Path::<PortableForm> {
-        segments: path.segments.iter().map(|e| e.ident.to_string()).collect(),
-    };
-    types.types.iter().any(|t| t.ty.path == scale_type_path)
+pub fn registry_contains_type_path(types: &PortableRegistry, path: &Vec<String>) -> bool {
```

Comment:
Nit: no need for the `Vec`

```suggestion
pub fn registry_contains_type_path(types: &PortableRegistry, path: &[String]) -> bool {
```